### PR TITLE
Drtii 1157 improve walking time predictions

### PR DESCRIPTION
**Pull request by richbirch**
Implement more flexible model training

### CODE REVIEW

File: src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/modeldefinitions/OffScheduleModelDefinition.scala [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    OneToMany(List(PartOfDay()), "pod"),
+    OneToMany(List(Carrier), "car"),
+    OneToMany(List(FlightNumber), "fln"),
+  )
+  override val aggregateValue: Arrival => Option[WithId] = TerminalOrigin.fromArrival
+  override val targetValueAndFeatures: Arrival => Option[(Double, Seq[String], Seq[Double])] = minutesOffSchedule(features)
+  override val baselineValue: Terminal => Double = (_: Terminal) => 0d
+}
+
+// Terminal/FlightNumber/Origin + day of week + part of day
```
Comment (ngu04) [inline src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/modeldefinitions/OffScheduleModelDefinition.scala]: I guess this comments are useful . May be some documention to understand model .

File: src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/modeldefinitions/ToChoxModelDefinition.scala [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    OneToMany(List(PartOfDay()), "pod"),
+    OneToMany(List(Origin), "ori"),
+    OneToMany(List(FlightNumber), "fln"),
+  )
+  override val aggregateValue: Arrival => Option[WithId] = TerminalCarrier.fromArrival
+  override val targetValueAndFeatures: Arrival => Option[(Double, Seq[String], Seq[Double])] = minutesToChox(features)
+  override val baselineValue: Terminal => Double = (_: Terminal) => defaultTimeToChox.toDouble
+}
+
+// Terminal/FlightNumber/Origin + day of week + part of day
```
Comment (ngu04) [inline src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/modeldefinitions/ToChoxModelDefinition.scala]: same here , I don't mind keeping it here as it make easier to understand what is happening

**Review by ngu04 — APPROVED** [review]
Excellent work !

File: src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/modeldefinitions/OffScheduleModelDefinition.scala (hunk 1 above)
Comment (richbirch) [reply src/main/scala/uk/gov/homeoffice/drt/analytics/prediction/modeldefinitions/OffScheduleModelDefinition.scala]: Yeah I won't keep these comments here as they really belong in the readme with more explanation

**Pull request merged**